### ISSUE TITLE
Urgent fixes needed for Waterfox 56.2! Please have a look.

### ISSUE THREAD

**Issue by Peacock365** [BrowserWorks/waterfox#567]
Hey there!

So, Waterfox 56.2 is out now. Fantastic! However, there are some fixes urgently required for this version, as some features are broken without them.

- _Problem no. 1 - The "About Waterfox" window (on Mac at least)_

"About Waterfox" does not display correctly on my Mac (macOS High Sierra). The window is completely white with a black ramification / frame around it. A window does open, but no actual information is displayed.

- _Problem no. 2 - The Fingerprinting changes._

I will just copy this from the gHacks user.js ver. 56, as I am not that versed with those particular settings:

> /* 4702: navigator.buildID (see gecko.buildID in about:config) reveals build time
> * down to the second which defeats user agent spoofing and can compromise OS etc
> * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=583181 ***/
   // user_pref("general.buildID.override", "20100101"); // (hidden pref)

> /* 4703: navigator.appName ***/
   // user_pref("general.appname.override", "Netscape"); // (hidden pref)
> /* 4704: navigator.appVersion ***/
   // user_pref("general.appversion.override", "5.0 (Windows)"); // (hidden pref)
> /* 4705: navigator.platform leaks in JS ***/
   // user_pref("general.platform.override", "Win32"); // (hidden pref)
> /* 4706: navigator.oscpu leaks in JS ***/
   // user_pref("general.oscpu.override", "Windows NT 6.1"); // (hidden pref)
> /* 4707: general.useragent.locale (related, see 0204) ***/

Just wanted to let you know.

-----------

Suggestion: Create a Waterfox 56.2.1 hotfix update that fixes these.

Thank you for your attention.

**Comment by Peacock365** [comment]
Problem no. 1 could be fixed via https://github.com/MrAlex94/Waterfox/pull/568

Would still need a new build to push to everyone though, and fix issue no. 2 beforehand.

**Comment by ghost** [comment]
_(Former)_ Problem no. 2 seems like, that it has already been fixed at 4368983

**Comment by Peacock365** [comment]
@Toromino

Nope. See here:

https://old.reddit.com/r/waterfox/comments/8k48rk/waterfox_5620_extensionscheckcompatibility/dz50k5m/?st=jhas7z3h&sh=5cba545b

**Comment by ghost** [comment]
@Peacock365 

Okay, so it is fixed, but @MrAlex94 forgot to include it in this release. So 4368983 should definitely be added to the next release!

**Comment by grahamperrin** [comment]
#538

**Comment by grahamperrin** [comment]
> Problem no. 3 - The Fingerprinting changes.

@jbeich re: https://www.reddit.com/comments/8k1p3j/-/dz5cv2a/?context=1 

With resistance enabled in 56.2.0 build 20180515213058, I get …`/Firefox/52.0` (not 60).   

```text
$ pkg info waterfox | grep Version
Version        : 56.2.0.7
```

Would you like this in FreeBSD Bugzilla? Or is 52.0 expected in the FreeBSD case? 

TIA

(For me personally, not urgent; I accept `privacy.resistFingerprinting` `false`.)

**Comment by jbeich** [comment]
@grahamperrin, because 1ad43082697b (unlike mozilla/gecko-dev@d20c7d929c84) makes less sense:
- WF56 supports fewer APIs than ESR60 (from [57](https://developer.mozilla.org/en-US/Firefox/Releases/57#New_APIs), [58](https://developer.mozilla.org/en-US/Firefox/Releases/58#New_APIs), [59](https://developer.mozilla.org/en-US/Firefox/Releases/59#New_APIs), [60](https://developer.mozilla.org/en-US/Firefox/Releases/60#New_APIs))
- WF56 supports fewer CSS properties than ESR60
- RFP in WF56 has more gaps compared to ESR60 (from [57](https://bugzilla.mozilla.org/buglist.cgi?quicksearch=blocked%3A1329996+milestone%3A57+status%3Aresolved), [58](https://bugzilla.mozilla.org/buglist.cgi?quicksearch=blocked%3A1329996+milestone%3A58+status%3Aresolved), [59](https://bugzilla.mozilla.org/buglist.cgi?quicksearch=blocked%3A1329996+milestone%3A59+status%3Aresolved), [60](https://bugzilla.mozilla.org/buglist.cgi?quicksearch=blocked%3A1329996+milestone%3A60+status%3Aresolved))
- WF56 is likely reach EOL together with ESR52

**Comment by Peacock365** [comment]
Just wanted to let you all know, I have deleted the issues relating to the Sync menu iOS link and add-on compatibility, as they are fixed already and a new build can be created which will have them fixed.

I have noticed a new issue on my Mac in relation to the "About Waterfox" window. This issue is my new no.1 in my initial post. Please have a look.

**Comment by Peacock365** [comment]
@jbeich:

Are you sure that Waterfox will be going EOL along with FF ESR 52? After all, Firefox 60 and Waterfox 56 are not that far apart. Wouldn't it be possible to apply most fixes, still? I understand that Waterfox 56 can't be kept up indefinitely due to web standard support, but I had at least expected it to be updated until late 2018 / early 2019... Sorry for going off-topic by the way, I am just interested in the matter.

**Comment by grahamperrin** [comment]
Thanks folks, I should steer the (non-urgent) APIs thread of this issue to: 

- [Webextension API merge strategy · Issue #389 …](https://github.com/MrAlex94/Waterfox/issues/389)

**Comment by grahamperrin** [comment]
> Problem no. 1 could be fixed via #568
> 
> Would still need a new build to push to everyone though, and fix issue no. 2 beforehand.

The numbers have changed, would you like to edit related comments? TIA. 

PS ideally with strike-through markup, so that newcomers to this issue can tell what problem numbers were before and after the change.

**Comment by Peacock365** [comment]
©grahamperrin:

No necessity for this, as the problems are already fixed, there is no need to further discuss them anyway. I do not intend for this issue to turn into a novel either. ;-)
